engine/resource/src/resource_archive.cpp

```diff
@@ -45,7 +45,20 @@ namespace dmResourceArchive
 
     int              g_NumArchiveLoaders = 0;
     ArchiveLoader    g_ArchiveLoader[4];
-    FDecryptResource g_ResourceDecryption = dmResourceArchive::DecryptWithXtea;
+
+
+    Result DecryptWithXtea(void* buffer, uint32_t buffer_len)
+    {
+        dmCrypt::Result cr = dmCrypt::Decrypt(dmCrypt::ALGORITHM_XTEA, (uint8_t*) buffer, buffer_len, (const uint8_t*) KEY, strlen(KEY));
+        if (cr != dmCrypt::RESULT_OK)
+        {
+            return RESULT_UNKNOWN;
+        }
+        return RESULT_OK;
+    }
+
+    FDecryptResource g_ResourceDecryption = DecryptWithXtea;
```

Comment:
We generally recommend defining global variables at the top of the file, for easier communication.

In this case, it would mean a forward declaration:

```
static Result DecryptWithXtea(void* buffer, uint32_t buffer_len);

FDecryptResource g_ResourceDecryption = DecryptWithXtea;
```